### ISSUE TITLE
Presentation compiler spurious error with cats, circe, slick, doobie, freek.... Anything Type involved

### ISSUE THREAD

**Issue by scabug** [scala/bug#9716]
I'm trying circe for Json, and the presentation compiler (that's the Scala IDE dev who said it :) is reporting false errors, as soon as I tried to do... well, almost anything related to circe jsons decoder :)

For example:
```scala
    // depency on circe 0.3.0, see: https://github.com/travisbrown/circe/
    import io.circe._
    import io.circe.generic.semiauto._
    import io.circe.parser._

    final case class User(email: String)

    object UserDecode {
      implicit lazy val userEventDecoder: Decoder[User] = deriveDecoder[User]
    }
```
The code compile, but on line 'implicit lazy val...', eclipse show a red-cross error under User:

```
Multiple markers at this line:
  not enough arguments for method deriveDecoder: (implicit decode: shapeless.Lazy[io.circe.generic.decoding.DerivedDecoder[nrm_marketing_data.User]])io.circe.Decoder[nrm_marketing_data.User]. Unspecified value parameter decode.
  could not find implicit value for parameter decode: shapeless.Lazy[io.circe.generic.decoding.DerivedDecoder[nrm_marketing_data.User]]
```

The same symptoms are appering with several cats methods, like traverseU.
Also: I don't have completion on anyting after the error, in the scope, for example in:

```scala
    ...... = {
    deriveDecoder[User].prepare( _.downField("user").downField("firstName")
    }
```

I don't have completion on prepare, and hitting crtl+space after ".p" leads to 0 scala completion.
Also, the type given on tooltip on mouse over is "val somevar: <error>"
So, in short, presentation compiler gave up at this point.

**Comment by fanf** [comment]
As already said in the https://github.com/tpolecat/doobie/issues/508, I can confirm that adding -Ymacro-expand:normal helps a lot regarding spurious errors. 

I also want to emphasis a behaviour of ScalaIDE that made me think at first that the option was not doing anything: in Scala IDE, when using ```right click on project -> scala -> set the scala installation```, Scala IDE actually change the per project settings config to "enable", and so your global command line args for scalac are no more used, and you need to copy/past them in ALL your projects dedicated properties. 

Hope it will help others.
